src/Phinx/Migration/AbstractMigration.php

```diff
@@ -370,4 +372,21 @@ public function shouldExecute(): bool
     {
         return true;
     }
+
+    /**
+     * Makes sure the version int is within range for valid datetime.
+     * This is required to have a meaningful order in the overview.
+     *
+     * @param int $version Version
+     * @return void
+     */
+    protected function validateVersion(int $version): void
+    {
+        $length = strlen((string)$version);
+        if (!$version || $length === 14) {
+            return;
+        }
```

Comment:
A follow-up here would be to validating that the various pieces of the string are all valid, e.g. that don't have `00` or `13` for `MM`.

Reply:
Sure, but then it wouldnt fit in a "minor" release probably as people could have existing invalid ones that are valid in range (sorting).
So for now I would like to keep it as is, but in the next major we could try to limit it to a more sane value in general.